feat(ReleaseUi): Add other / detected license in release

akapti opened the pull request:
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
 * PR to display other license from CLI file and add it to `other license` attribute of `release`.
 * Which issue is this pull request belonging to and how is it solving it? #1051 
 * Did you add or update any new dependencies that are required for your change? - `No`

Issue: #1051 

### Suggest Reviewer
> You can suggest reviewers here with an @mention.

### How To Test?
> How should these changes be tested by the reviewer?
 * Attach CLI files with Main/Other license ids to a release.
 * Go to `Clearing Details` tab in `Release Details` page, and click on `Show License Info` button in `SPDX Attachments` table.
 * Clicking on `Show License Info` button should load all the licenses from the CLI.
 * Click on `Add data to this release` button to add the Main/License ids to the release.
> Have you implemented any additional tests? - `No`

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR


Signed-off-by: Abdul Kapti <abdul.kapti@siemens-healthineers.com>

smrutis1 commented on frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java:
```diff
@@ -587,10 +589,26 @@ private void loadSpdxLicenseInfo(ResourceRequest request, ResourceResponse respo
                 });
                 jsonGenerator.writeEndArray();
-            } else if (licenseWithText != null) {
+            } else if (mainLicenseNames.size() > 0) {
```
`CommonUtils.isNotEmpty(mainLicenseNames)` would look good here.

akapti replied:
added the requested change.

smrutis1 commented on frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java:
```diff
@@ -587,10 +589,26 @@ private void loadSpdxLicenseInfo(ResourceRequest request, ResourceResponse respo
-                jsonGenerator.writeArrayFieldStart("licenseIds");
-                jsonGenerator.writeString(licenseWithText.getLicenseSpdxId());
+                jsonGenerator.writeArrayFieldStart(LICENSE_IDS);
+                mainLicenseNames.forEach(license -> {
```
May be `wrapException(_)` can be used here.

akapti replied:
done

smrutis1 commented on frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java:
```diff
@@ -587,10 +589,26 @@ private void loadSpdxLicenseInfo(ResourceRequest request, ResourceResponse respo
+                jsonGenerator.writeStringField("otherLicense", LanguageUtil.get(resourceBundle,"other.license.id"));
+                jsonGenerator.writeArrayFieldStart("otherLicenseIds");
+                otherLicenseNames.forEach(license -> {
+                    try {
```
`wrapException(_)` here too?

akapti replied:
done.

smrutis1 reviewed (changes requested):
few minor comments.

smrutis1 commented:
Tested it as per the mentioned steps successfully. However observed one issue(Seems to be there in current master too), clicking on the added license on release summary page redirects to component summary page.

smrutis1 commented:
Some UI issues while editing.
![image](https://user-images.githubusercontent.com/46560890/104994584-c094f780-5a4a-11eb-81e8-7fab56b1f135.png)

akapti commented:
> Some UI issues while editing.

Fixed UI issues and incorporated the code review changes.

smrutis1 reviewed (approved):
Code looks good.

akapti commented:
From Discussion: Removed License link from `Component` and `Release` details page.

JaideepPalit commented:
Seems to be some issues. - 
- License Link in Project Clearing status, seems to be redirecting to Project Details Page.
- For some spdx file (shared separately), getting below error on `Add data to release`
![image](https://user-images.githubusercontent.com/56534710/106265632-56036900-624d-11eb-8b08-c7a56ec2a13d.png)

> From Discussion: Removed License link from `Component` and `Release` details page.
 - working fine

akapti commented:
> * License Link in Project Clearing status, seems to be redirecting to Project Details Page.

Removed License link for all the  pages where wrong link was being generated.

> * For some spdx file (shared separately), getting below error on `Add data to release`

Fixed the issue.

JaideepPalit reviewed (approved):
Tested successfully
